syft/source/location.go

```diff
@@ -46,22 +45,11 @@ func NewLocationFromCoordinates(coordinates Coordinates) Location {
 
 // NewLocationFromImage creates a new Location representing the given path (extracted from the ref) relative to the given image.
 func NewLocationFromImage(virtualPath string, ref file.Reference, img *image.Image) Location {
-	entry, err := img.FileCatalog.Get(ref)
-	if err != nil {
-		log.Warnf("unable to find file catalog entry for ref=%+v", ref)
-		return Location{
-			Coordinates: Coordinates{
-				RealPath: string(ref.RealPath),
-			},
-			VirtualPath: virtualPath,
-			ref:         ref,
-		}
-	}
-
+	layer := img.FileCatalog.Layer(ref)
```

Comment:
Is there any chance layer or its fields `Metadata` could be nil here and panic?

Reply:
I don't think so, no